build/build.go

```diff
@@ -1478,26 +1481,53 @@ func LoadInputs(ctx context.Context, d driver.Driver, inp Inputs, pw progress.Wr
 		// handle OCI layout
 		if strings.HasPrefix(v.Path, "oci-layout://") {
 			pathAlone := strings.TrimPrefix(v.Path, "oci-layout://")
-			parts := strings.SplitN(pathAlone, "@", 2)
-			if len(parts) != 2 {
-				return nil, errors.Errorf("invalid oci-layout context %s, must be oci-layout:///path/to/layout@sha256:hash", v.Path)
+			localPath := pathAlone
+			localPath, dig, hasDigest := strings.Cut(localPath, "@")
+			localPath, tag, hasTag := strings.Cut(localPath, ":")
+			if !hasDigest {
+				if !hasTag {
+					tag = "latest"
```

Comment:
If `index` only contains one record then the `tag==latest` constraint shouldn't be required.

Reply:
Have added another branch to check for this case.